--- files/MiSTer_example.ini	2024-03-25 15:34:27.511069700 +0000
+++ files/MiSTer_example.ini	2026-08-07 06:35:36.179953700 +0000
@@ -1,11 +1,11 @@
 [MiSTer]
-;debug=1               ; set to 1 to enable debugging messages. Default is 0(disabled).
+;sanity_check=0        ; ini sanity check. Set to 1 to report about INI errors on OSD such as invalid options.
+;debug=1               ; set to 1 to enable debugging messages, 2 - write to /tmp/debug.txt. Default is 0(disabled).
 key_menu_as_rgui=0     ; set to 1 to make the MENU key map to RGUI in Minimig (e.g. for Right Amiga)
 forced_scandoubler=0   ; set to 1 to run scandoubler on VGA output always (depends on core).
-;ypbpr=0               ; set to 1 for YPbPr on VGA output. (obsolete. see vga_mode)
-vga_mode=rgb           ; supported modes: rgb, ypbpr, svideo, cvbs. rgb is default.
+vga_mode=rgb           ; supported modes: rgb, ypbpr, svideo, cvbs, and subcarrier (for external rgb converters). rgb is default.
 ntsc_mode=0            ; Only for S-Video and CVBS vga_mode. 0 - normal NTSC, 1 - PAL-60, 2 - PAL-M.
-composite_sync=0       ; set to 1 for composite sync on HSync signal of VGA output.
+composite_sync=1       ; set to 1 for composite sync on HSync signal of VGA output.
 vga_scaler=0           ; set to 1 to connect VGA to scaler output.
 hdmi_audio_96k=0       ; set to 1 for 96khz/16bit HDMI audio (48khz/16bit otherwise)
 keyrah_mode=0x18d80002 ; VIDPID of keyrah for special code translation (0x23418037 for Arduino Micro)
@@ -22,7 +22,17 @@
 menu_pal=0             ; 1 - PAL mode for menu core
 hdmi_limited=0         ; 1 - use limited (16..235) color range over HDMI
                        ; 2 - use limited (16..255) color range over HDMI, for VGA converters.
+hdmi_cec=0             ; 1 - enable HDMI-CEC via ADV7513, 0 - disable.
+hdmi_cec_input_mode=1  ; 0 - ignore CEC remote input, 1 - accept mapped CEC keys.
+hdmi_cec_power_on=1    ; 1 - on power-on, try once to make MiSTer the active HDMI input once EDID provides its CEC address.
+hdmi_cec_sleep=0       ; 1 - send CEC STANDBY after video_off minutes of no input activity. Requires video_off>0.
+hdmi_cec_wake=0        ; 1 - on input after that idle period, send wake (ImageViewOn/TextViewOn/ActiveSource).
+hdmi_cec_clock=0       ; 0 - auto-detect 12MHz/50MHz, otherwise floating point value from 3.0 to 100.0 (3-100MHz).
+                       ;
 direct_video=0         ; 1 - enable core video timing over HDMI, use only with VGA converters.
+                       ; 2 - auto-mode for HDMI DACs that report 1024x768 resolution (AG6200, CS5213). 
+                       ; Power cycle when switching between HDMI displays and DACs. No hot-plug detection.
+
 hdr=0                  ; 1 - enable HDR using HLG (recommended for most users)
                        ; 2 - enable HDR using the DCI P3 color space (use color controls to tweak, suggestion: set saturation to 80).
 fb_size=0              ; 0 - automatic, 1 - full size, 2 - 1/2 of resolution, 4 - 1/4 of resolution.
@@ -30,8 +40,10 @@
 osd_timeout=30         ; 5-3600 timeout (in seconds) for OSD to disappear in Menu core. 0 - never timeout.
                        ; Background picture will get darker after double timeout
 video_off=0            ; output black frame in Menu core after timeout (is seconds). Valid only if osd_timout is non zero.
+video_off_logo=0       ; 1 - show moving MiSTer logo when video_off is enabled (to distinguish from TV off mode)
 osd_rotate=0           ; Display OSD menu rotated,  0 - no rotation, 1 - rotate right (+90°), 2 - rotate left (-90°)                  
 vga_sog=0              ; 1 - enable sync on green (needs analog I/O board v6.0 or newer).
+lookahead=1            ; 1 - keep cursor centered while scrolling, 0 - only scroll at list edges
 
 
 ; 1 - enables the recent file loaded/mounted.
@@ -55,7 +67,7 @@
 ;  1136 bytes - chars 0-141
 ;  up to 2048 - only chars 0-141 will be used.
 ; if first 32 chars are empty (for sizes 1024 bytes and more) then they are skipped.
-font=font/myfont.pf
+;font=font/myfont.pf
 
 ; USER button emulation using a keyboard. Usually it's the reset button.
 ; 0 - lctrl+lalt+ralt (lctrl+lgui+rgui on keyrah)
@@ -109,10 +121,19 @@
 ; modes as a base even for 50Hz systems. 
 vsync_adjust=0
 
+; Variable Refresh Rate control
+; 0 - Do not enable VRR (send no VRR control frames)
+; 1 - Auto Detect VRR from display EDID.
+; 2 - Force Enable Freesync
+; 3 - Force Enable VESA VRR
+; 4 - Force Enable MiSTer VRR (vsync_adjust must be 0)
+vrr_mode=0
+; VESA VRR base framerate. Normally set to the current video mode's output framerate
+vrr_vesa_framerate=0
+
 ; If your monitor doesn't support either very low (NTSC monitors may not support PAL) or 
 ; very high (PAL monitors may not support NTSC) then you can set refresh_min and/or refresh_max
-; parameters, so vsync_adjust won't be applied for refreshes outside specified.
-; These parameters are valid only when vsync_adjust is non-zero.
+; parameters, so neither vsync_adjust nor VRR will be applied for refreshes outside specified.
 refresh_min=0
 refresh_max=0
 
@@ -321,22 +342,16 @@
 ; can give worse result on others. Default is 0 (non-game).
 ;hdmi_game_mode=1
 
-; Variable Refresh Rate control
-; 0 - Do not enable VRR (send no VRR control frames)
-; 1 - Auto Detect VRR from display EDID. 
-; 2 - Force Enable Freesync
-; 3 - Force Enable Vesa HDMI Forum VRR
-vrr_mode=0
-; Minimum framerate in VRR mode. 
-vrr_min_framerate=0
-; Maximum framerate in VRR mode (currently only used in Freesync mode). 
-vrr_max_framerate=0
-; VESA VRR base framerate. Normally set to the current video mode's output framerate
-vrr_vesa_framerate=0
-
 ; disable autofire if for some reason it's not required and accidentally triggered
 disable_autofire=0
 
+; allow autofire on directional inputs. typically not recommended. default is 0 (disabled).
+;autofire_on_directions=1
+
+; custom autofire rates (in hertz). will round to nearest rate that divides evenly into 60hz.
+; can also use literal bit patterns for on/off cycles, i.e. 0b00111. each bit is one frame.
+;autofire_rates=10,0b00111,20,30
+
 ; Specify a default video processing preset that will be applied to cores.
 ; Path is relative to the presets/ directory and can optionally include the .ini extension
 ;preset_default=General Hardware/Console - 3rdGen
@@ -364,3 +379,29 @@
 
 ; use custom main for specific core. This option should be used only inside specific core.
 ;main=some_binary_file
+
+; Code of button on XBox Elite 2 controller used as a special shift button to emulate keys absent
+; on chatpad (and remapped through XBox accessory app buttons as a keys).
+; XBE2 controller with chatpad accessory and this option can replace full(almost) keyboard and mouse.
+; Note: chatpad works only when XBE2 is connected through cable or USB dongle. Not through Bluetooth.
+;xbe2_shift=0x224 ;Paddle P3
+
+; Some board have i2c communication problem between FPGA and HDMI chip making OSD (and even core itself) laggy.
+; This option reduces amount of i2c communication by removing SPD packet change from specific events.
+; Note: SPD packets are most important if you use Direct Video mode together with external scaler which
+; supports MiSTer specifically. If you don't use such scaler or don't use Direct Video mode then it's safe to set 
+; this option to 3
+; 0 - normal SPD work (default).
+; 1 - Don't send SPD in normal mode (non-Direct-Video).
+; 2 - Additionally don't send new SPD upon OSD open/close
+; 3 - Don't send SPD at all.
+;spd_quirk=0
+
+; 1...1440 - minutes till HDMI off if no user activity detected (no input from mouse/keyboard/joystick).
+; unlike video_off which works only in Menu core, this option works in all cores.
+; 0 (default) - don't turn HDMI off.
+hdmi_off=0
+
+; Use specific keyboard(s) as a joystick. Multiple entries are supported (one per line).
+; format is 0xVIDPID
+;keyboard_as_joystick=0x12345678
